src/assisted_test_infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -599,6 +600,40 @@ def set_ipxe_url(self, network_name: str, ipxe_url: str):
         self.destroy_network(network)
         self.create_network(xml.toprettyxml())
 
+    def _create_dns_host_xml(self, api_vip: str, cluster_name: str):
+        """
+        Creates an XML objet for adding a DNS host to a libvirt network
+        """
+        host_element = Element("host", ip=api_vip)
+        if cluster_name != "":
+            hostname = f"api.{cluster_name}.{self._entity_config.base_dns_domain}"
+            hostname_element = SubElement(host_element, "hostname")
+            hostname_element.text = hostname
+        return minidom.parseString(tostring(host_element))
+
+    def add_dns_host_to_network(self, network_name: str, api_vip: str, cluster_name: str):
+        """
+        Adds a dns-host to an existing network
+        """
+        host_xml = self._create_dns_host_xml(api_vip, cluster_name)
+        network = self.get_network_by_name(network_name)
+        network.update(
+            libvirt.VIR_NETWORK_UPDATE_COMMAND_ADD_LAST,
+            libvirt.VIR_NETWORK_SECTION_DNS_HOST,
+            -1,
+            host_xml.toprettyxml(),
+        )
+
+    def remove_dns_host_from_network(self, network_name: str, api_vip: str, cluster_name: str):
```

Comment:
in general I prefer those kind of stuff to be immutable rather than mutable. Is there a need for mutability here?

Reply:
Sorry I need a little bit of clarification: do you mean that once the dns-host is added to the network, then it shouldn't be removed?

Reply:
Unless I'm missing something (adding the usage can help here clarifying the use-case), if the host stays permanent during that test, then yes. the DNS doesn't have to be removed
cleanup afterwards should remove all relevant configuration just by destroying the libvirt hosts and libvirt networks

Reply:
Gotcha! So these functions are meant for the assisted-service libvirt network (aka `default` network), not the spoke cluster's network. 

Here's an example usage 
https://github.com/openshift/assisted-test-infra/pull/1816/files#diff-21e688d6eb3b87727264f6934118765f035f5bf7504dff17b3ab9fb388f4d054R415-R422

The reason we'd want a separate function for removing it is because IP could be used (re-used) for another spoke cluster if the original spoke cluster is removed. Does that make sense?

Reply:
okay. I missed it happening on the ``default`` network and not other libvirt networks
technically speaking assisted-service should reside in the ``minikube`` network, but I guess everything configured on the ``default`` network propagates to all other libvirt networks so it catches it as well

As for the mutability, if it's happening for the ``default`` network it means we must have ``remove_dns_host_from_network`` function, because nothing will clean that up. So that's not an issue